packages/components/src/QueryModel/withQueryModels.tsx

```diff
@@ -656,8 +656,11 @@ export function withQueryModels<Props>(
                 produce((draft: Draft<State>) => {
                     // Instantiate the model first because queryConfig.id is optional and is auto-generated in the
                     // QueryModel constructor if not set.
-                    const queryModel = new QueryModel(queryConfig);
+                    let queryModel = new QueryModel(queryConfig);
```

Comment:
Since this is repeated in initModels, should QueryModel constructor takes an optional location parameter and handle the processing instead?